packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -1447,18 +1447,23 @@ function build(
     } else if (delivery[0] === 'stopVat') {
       return meterControl.runWithoutMeteringAsync(stopVat);
     } else {
+      let complete = false;
       // Start user code running, record any internal liveslots errors. We do
       // *not* directly wait for the userspace function to complete, nor for
       // any promise it returns to fire.
       const p = Promise.resolve(delivery).then(unmeteredDispatch);
-
-      // Instead, we wait for userspace to become idle by draining the promise
-      // queue. We clean up and then return 'p' so any bugs in liveslots that
-      // cause an error during unmeteredDispatch will be reported to the
-      // supervisor (but only after userspace is idle).
+      p.finally(() => (complete = true));
+
+      // Instead, we wait for userspace to become idle by draining the
+      // promise queue. We clean up and then examine/return 'p' so any
+      // bugs in liveslots that cause an error during
+      // unmeteredDispatch (or a 'buildRootObject' that fails to
+      // complete in time) will be reported to the supervisor (but
+      // only after userspace is idle).
       return gcTools.waitUntilQuiescent().then(() => {
         afterDispatchActions();
-        return p;
+        // eslint-disable-next-line prefer-promise-reject-errors
+        return complete ? p : Promise.reject('buildRootObject unresolved');
```

Comment:
The not `complete` rejection is technically not `startVat`/`buildRootObject` specific, right? In practice it doesn't happen for other deliveries, but I'm a little concerned about using a specific call out here as it may be confusing to readers.

Reply:
Correct.. the only dispatch that pays attention to a userspace-provided Promise is `startVat`, so the only place that can cause `!complete` is `buildRootObject`, so the error message is specialized. I'll add a comment to that effect.